dlc-manager/src/contract_updater.rs

```diff
@@ -457,12 +484,15 @@ where
     )?;
 
     let dlc_transactions = DlcTransactions {
-        fund,
+        fund: fund.clone(),
         cets,
         refund: refund.clone(),
         funding_script_pubkey: funding_script_pubkey.clone(),
     };
 
+    // check that we have unsigned version of funding tx so we can create the PSBT
+    debug_assert!(fund.input.iter().all(|i| i.witness.is_empty()));
```

Comment:
Not sure why this is here? (and if you remove that you can remove the `clone` above)

Reply:
We need the clone because `fund` is a reference.